### DISCUSSION TITLE
Check for VBF jets

### DISCUSSION THREAD

**Discussion by ram1123** [ram1123/2l2Q_limitSettingTool#29] [Comments_HIG_23_001] [answered]
- [x] GEN Level plot: 
   - [x] Plot the pdgID of all quarks at the GEN level. I mean the histogram of PDGID, where the x-axis will be PDGID
   - [x] Plot the pdgID of VBF type quark at the GEN level. I mean the histogram of PDGID, where the x-axis will be PDGID
   - [x] Plot 2D distribution (eta vs pT) of leading in pT VBF type quark (with finer bins).
   - [x] Plot 2D distribution (eta vs pT) of sub-leading in pT VBF type quark (with finer bins).
- [x] RECO level plots
   - [x] Plot nJets distribution without any selection
   - [x] Plot nJets distribution with leptonic selection but without any cuts on JET only
   - [x] Plot nJets distribution after final selection in merged and resolved category, separately.

**Comment by jialin-guo1** [comment]
|Description | Plot |
|--- |--- |
|nJets distribution without any selection| ![image](https://github.com/ram1123/2l2Q_limitSettingTool/assets/68882142/11a7612e-12f4-48ef-b6a1-8065ebceeaef)|
|nJets distribution with leptonic selection but without any cuts on JET only| ![image](https://github.com/ram1123/2l2Q_limitSettingTool/assets/68882142/c330bc29-324a-4663-9c72-1f61b351f5d0)|
|nJets distribution after final selection in resolved category| ![image](https://github.com/ram1123/2l2Q_limitSettingTool/assets/68882142/c0d6c2ca-168e-4fb3-a326-2ac81a854a1e)
|nJets distribution after final selection in merged category | ![image](https://github.com/ram1123/2l2Q_limitSettingTool/assets/68882142/5d94c523-0bb2-413a-9ae9-88f88f6718ee)|



Notation: all plots are drawn under pt(jet) cut on 10GeV

**Reply by ram1123** [reply]
Can you try to plot all plots on one canvas, so that its easy to compare?

**Reply by jialin-guo1** [reply]
![image](https://github.com/ram1123/2l2Q_limitSettingTool/assets/68882142/d1057759-52d3-4ce4-8980-511fd0e50fac)


**I found that 2 tight lepton selection have already implemented in Ntuplzer, I have to start from miniAOD to get information without any selection**

**Comment by jialin-guo1** [comment]
**pdgID of all quarks at the GEN level and VBF type quark **

![image](https://github.com/ram1123/2l2Q_limitSettingTool/assets/68882142/144ccbcb-518a-4b50-9155-d9d8f7955655)

**Reply by ram1123** [reply]
Something wrong. PDG ID should not be more than +/- 6.

**Reply by jialin-guo1** [reply]
![image](https://github.com/ram1123/2l2Q_limitSettingTool/assets/68882142/442064f6-d559-457b-9706-16d80572fcb0)

I checked the PDG ID list. I don't think we will get PDG iD great than 6 for VBF quark

**Reply by ram1123** [reply]
I am not saying that you will get more than 6. 

I am asking why you have something after 6?

**Comment by jialin-guo1** [comment]
![image](https://github.com/ram1123/2l2Q_limitSettingTool/assets/68882142/388daf4e-243d-4055-abe6-cba9ec60ef39)
![image](https://github.com/ram1123/2l2Q_limitSettingTool/assets/68882142/dee1bc94-5e5c-4480-b582-ba92cdd83386)

**eta vs pt with pt leading and pt sub-leading**

**Reply by ram1123** [reply]
Please cross-check if you plotted the leading and sub-leading separately.

As I don't expect that all leading pT quarks will have positive pseudo-rapidity and the sub-leading in a negative pseudo-rapidity region.

**Reply by jialin-guo1** [reply]
yes. you are right. now I am re-plotting them

**Reply by jialin-guo1** [reply]
![image](https://github.com/ram1123/2l2Q_limitSettingTool/assets/68882142/e9793184-c1f0-459a-a65e-0a36ccfc8ab9)

this this leading pt vs eta

**Reply by jialin-guo1** [reply]
![image](https://github.com/ram1123/2l2Q_limitSettingTool/assets/68882142/e8aef988-3d20-4e1a-bcf9-71f954acaa4c)

This is subleasing pt VS eta

**Reply by ram1123** [reply]
Good. But, it would be better if you can get the two with updated range of (0, 150) GeV.

**Reply by jialin-guo1** [reply]
![image](https://github.com/ram1123/2l2Q_limitSettingTool/assets/68882142/7c6351ec-a88d-4839-8e13-609e1952d370)

I only update pt range to 0-150Gev for subleasing jet. because there still many events with pt larger than 200GeV

**Comment by jialin-guo1** [comment]
<img width="992" alt="image" src="https://github.com/ram1123/2l2Q_limitSettingTool/assets/68882142/ef00afc7-530e-41be-bec7-2fe80259ae0f">

**Reply by ram1123** [reply]
Hi @ram1123 ,

You should have 2 VBF category in new table no?

**Reply by jialin-guo1** [reply]
no event who passed KD_jjVBF or KD_jVBF will category to VBF. So, in the end, we still only have two categories  ggh and VBF

**Reply by ram1123** [reply]
You have the KD distribution for >= 4 jets and >= 3 jets.

Can you plot again >= 3 jets after removing the events that belong to >= 4 jets? Your current plot for> = 3 jets is the inclusive plot. Right?

I am asking this as I still see too many events falling in the ggH category.

**Reply by jialin-guo1** [reply, marked as answer]
**Can you plot again >= 3 jets after removing the events that belong to >= 4 jets? Your current plot for> = 3 jets is the inclusive plot. Right?**

-->in these two plots. Now, I compute KD_2jet only when njets>=4 in event. and if njets==3, I compute KD_1jet, otherwise, njet<=2, belong to ggH
<img width="450" alt="image" src="https://github.com/ram1123/2l2Q_limitSettingTool/assets/68882142/e00ffe0b-0460-41ee-ab95-0c0bed213d33">
<img width="450" alt="image" src="https://github.com/ram1123/2l2Q_limitSettingTool/assets/68882142/ae624125-8621-4d69-82bf-4422eb80a461">

we still have events with exactly njets==2 will category to ggH 
<img width="530" alt="image" src="https://github.com/ram1123/2l2Q_limitSettingTool/assets/68882142/f7282ffc-f803-4697-8b90-87e6efbba235">

**Reply by ram1123** [reply]
Okay. So, not good separation for nJets == 3. 

Share the root files when its there in the public.

**Reply by jialin-guo1** [reply]
/eos/user/g/guoj/Sample/2L2Q/UL_Legacy/2018

**Comment by ram1123** [comment]
> **Can you plot again >= 3 jets after removing the events that belong to >= 4 jets? Your current plot for> = 3 jets is the inclusive plot. Right?**
> 
> -->in these two plots. Now, I compute KD_2jet only when njets>=4 in event. and if njets==3, I compute KD_1jet, otherwise, njet<=2, belong to ggH <img alt="image" width="450" src="https://private-user-images.githubusercontent.com/68882142/255926510-e00ffe0b-0460-41ee-ab95-0c0bed213d33.png?jwt=eyJhbGciOiJIUzI1NiIsInR5cCI6IkpXVCJ9.eyJpc3MiOiJnaXRodWIuY29tIiwiYXVkIjoicmF3LmdpdGh1YnVzZXJjb250ZW50LmNvbSIsImtleSI6ImtleTEiLCJleHAiOjE2OTIyOTY5NzksIm5iZiI6MTY5MjI5NjY3OSwicGF0aCI6Ii82ODg4MjE0Mi8yNTU5MjY1MTAtZTAwZmZlMGItMDQ2MC00MWVlLWFiOTUtMGMwYmVkMjEzZDMzLnBuZz9YLUFtei1BbGdvcml0aG09QVdTNC1ITUFDLVNIQTI1NiZYLUFtei1DcmVkZW50aWFsPUFLSUFJV05KWUFYNENTVkVINTNBJTJGMjAyMzA4MTclMkZ1cy1lYXN0LTElMkZzMyUyRmF3czRfcmVxdWVzdCZYLUFtei1EYXRlPTIwMjMwODE3VDE4MjQzOVomWC1BbXotRXhwaXJlcz0zMDAmWC1BbXotU2lnbmF0dXJlPTZiY2NiMDY4Y2UzZmI2YWM2ZDliOGU2ZjQ1MThkM2MzMTMwYWU2Y2Q5Nzk2MjQ4Y2FlYmQwYmM1NTM5MTJmNjQmWC1BbXotU2lnbmVkSGVhZGVycz1ob3N0JmFjdG9yX2lkPTAma2V5X2lkPTAmcmVwb19pZD0wIn0.g0YixZ_CHS1fCn49pOEhnNFybb-Q-nIj_D_SGEDc-7g"> <img alt="image" width="450" src="https://private-user-images.githubusercontent.com/68882142/255926533-ae624125-8621-4d69-82bf-4422eb80a461.png?jwt=eyJhbGciOiJIUzI1NiIsInR5cCI6IkpXVCJ9.eyJpc3MiOiJnaXRodWIuY29tIiwiYXVkIjoicmF3LmdpdGh1YnVzZXJjb250ZW50LmNvbSIsImtleSI6ImtleTEiLCJleHAiOjE2OTIyOTY5NzksIm5iZiI6MTY5MjI5NjY3OSwicGF0aCI6Ii82ODg4MjE0Mi8yNTU5MjY1MzMtYWU2MjQxMjUtODYyMS00ZDY5LTgyYmYtNDQyMmViODBhNDYxLnBuZz9YLUFtei1BbGdvcml0aG09QVdTNC1ITUFDLVNIQTI1NiZYLUFtei1DcmVkZW50aWFsPUFLSUFJV05KWUFYNENTVkVINTNBJTJGMjAyMzA4MTclMkZ1cy1lYXN0LTElMkZzMyUyRmF3czRfcmVxdWVzdCZYLUFtei1EYXRlPTIwMjMwODE3VDE4MjQzOVomWC1BbXotRXhwaXJlcz0zMDAmWC1BbXotU2lnbmF0dXJlPTg1YzgyZTk3YjIxYWJlMDhjN2EzMTM2Y2IwMGZkMzU1NjJhNmQwOWQwZGE5YjE5ZTljZTMyZTA2M2NkNDZjMWUmWC1BbXotU2lnbmVkSGVhZGVycz1ob3N0JmFjdG9yX2lkPTAma2V5X2lkPTAmcmVwb19pZD0wIn0.n3F2ros6-IgGHTKO4NS4prsMOLHd2Mdt7q_2aThJ2wg">
> 
> we still have events with exactly njets==2 will category to ggH <img alt="image" width="530" src="https://private-user-images.githubusercontent.com/68882142/255929212-f7282ffc-f803-4697-8b90-87e6efbba235.png?jwt=eyJhbGciOiJIUzI1NiIsInR5cCI6IkpXVCJ9.eyJpc3MiOiJnaXRodWIuY29tIiwiYXVkIjoicmF3LmdpdGh1YnVzZXJjb250ZW50LmNvbSIsImtleSI6ImtleTEiLCJleHAiOjE2OTIyOTY5NzksIm5iZiI6MTY5MjI5NjY3OSwicGF0aCI6Ii82ODg4MjE0Mi8yNTU5MjkyMTItZjcyODJmZmMtZjgwMy00Njk3LThiOTAtODdlNmVmYmJhMjM1LnBuZz9YLUFtei1BbGdvcml0aG09QVdTNC1ITUFDLVNIQTI1NiZYLUFtei1DcmVkZW50aWFsPUFLSUFJV05KWUFYNENTVkVINTNBJTJGMjAyMzA4MTclMkZ1cy1lYXN0LTElMkZzMyUyRmF3czRfcmVxdWVzdCZYLUFtei1EYXRlPTIwMjMwODE3VDE4MjQzOVomWC1BbXotRXhwaXJlcz0zMDAmWC1BbXotU2lnbmF0dXJlPWFlMzJmYzVmOWY1MGU0MTdhN2JlM2YzNGEzMDk0MGUyNzkyNGE2YjY1NGM1NWI1NDI3NDU5ZmRhNTIyNWIzNzgmWC1BbXotU2lnbmVkSGVhZGVycz1ob3N0JmFjdG9yX2lkPTAma2V5X2lkPTAmcmVwb19pZD0wIn0.eO1Sm6aap6Ej2AnyZJ2nQC6zVuGBJucoNIuDEu2S8ng">

Here the plots that you added is for the resolved category. Please add the plots for the merged category too.

**Reply by jialin-guo1** [reply]
the plots is broken. I can't see them

**Reply by ram1123** [reply]
refresh the page it should work. Its the plot which is marked as answer.

**Reply by jialin-guo1** [reply]
merged
![image](https://github.com/ram1123/2l2Q_limitSettingTool/assets/68882142/885f5442-ce91-4655-b0c1-3c04b1a7088d)
![image](https://github.com/ram1123/2l2Q_limitSettingTool/assets/68882142/5748ad23-5294-4f93-bfca-feb2a5bf6550)
